src/hdmint/wallet.cpp

```diff
@@ -324,22 +315,29 @@ bool CHDMintWallet::SeedToZerocoin(const uint512& seedZerocoin, GroupElement& co
     return true;
 }
 
-CKeyID CHDMintWallet::GetZerocoinSeedID(uint32_t n){
+CKeyID CHDMintWallet::GetZerocoinSeedData(int32_t nCount, uint160 hashSeedMaster){
     // Get CKeyID for n from mintpool
-    std::pair<CKeyID,uint32_t> mintPoolEntry;
-    if(!mintPool.Get(n, mintPoolEntry)){
+    uint256 hashPubcoin;
+    std::pair<uint256,MintPoolEntry> mintPoolEntryPair;
+
+    while(!mintPool.Get(nCount, hashSeedMaster, mintPoolEntryPair)){
         // Top up mintpool if empty
         GenerateMintPool();
-        mintPool.Get(n, mintPoolEntry);
     }
 
-    return mintPoolEntry.first;
+    return get<1>(mintPoolEntryPair.second);
 }
 
-uint512 CHDMintWallet::GetZerocoinSeed(uint32_t n, CKeyID& seedId)
+uint512 CHDMintWallet::CreateZerocoinSeed(int32_t& n, CKeyID& seedId, bool checkIndex)
```

Comment:
If we do not modify n, please do not pass by reference.

Reply:
Actually I except something like `int32_t n`.

Reply:
It's passing by const reference now?